src/setup.ts

```diff
@@ -58,7 +57,17 @@ export function mixin(Vue: VueConstructor) {
     }
 
     Object.keys(binding).forEach(name => {
-      const bindingValue = binding[name];
+      let bindingValue = binding[name];
+      if (bindingValue === undefined)
+        return;
+      // make plain value reactive
+      if (!isWrapper(bindingValue) && !isFunction(bindingValue) && !Object.isFrozen(bindingValue)) {
+        if (isPlainObject(bindingValue))
+          bindingValue = state(bindingValue);
```

Comment:
We should always use `value`.  Consider the example below. The assignment to `form` in the template will not trigger a re-render if we use `state`.

```js
{
  setup() {
    return {
      form: { a:'', b: ''}
    }
  },
   template: `
      <div>
        a: {{ form.a }} b: {{ form.b }}
        <button @click="form = { a: 1, b: 2}"><button>
      <div>
    `
}
```

Reply:
I have updated the test in 950caa1ba01278d21b654a9b49eb3a707a91787f
```js
  it('should make returned plain value reactive', done => {
    const vm = new Vue({
      setup() {
        return {
          name: null,
          nested: {
            object: {
              msg: 'foo'
            }
          },
          foo: {
            bar: 0
          }
        };
      },
      template: '<div>{{ name }}, {{ nested.object.msg }}, {{ foo.bar }}</div>',
    }).$mount();
    expect(vm.$el.textContent).toBe(', foo, 0');
    vm.name = 'foo';
    vm.nested.object.msg = 'bar';
    vm.foo = { bar: 1 };
    waitForUpdate(() => {
      expect(vm.$el.textContent).toBe('foo, bar, 1');
    }).then(done);
  })
```
Both `state` and `value` approaches seems passed the test
```ts
      // make plain value reactive
      if (!isWrapper(bindingValue) && !isFunction(bindingValue) && !Object.isFrozen(bindingValue)) {
        if (isPlainObject(bindingValue))
          bindingValue = state(bindingValue);
        else
          bindingValue = value(bindingValue);
      }
```
```ts
      // make plain value reactive
      if (!isWrapper(bindingValue) && !isFunction(bindingValue) && !Object.isFrozen(bindingValue)) {
        bindingValue = value(bindingValue);
      }
```
I am not sure whether these two approaches have difference in performance on re-rendering or not. 
I would use `value` alone for now.

Reply:
```js
// these can trigger the re-render
vm.name = 'foo';
vm.nested.object.msg = 'bar';

// this can't
vm.foo = { bar: 1 };
```

Change the test to: 
```js
  it('should make returned plain value reactive', done => {
    const vm = new Vue({
      setup() {
        return {
          foo: {
            bar: 0
          }
        };
      },
      template: '<div>{{ foo.bar }}</div>',
    }).$mount();
    expect(vm.$el.textContent).toBe('0');
    vm.foo = { bar: 1 };
    waitForUpdate(() => {
      expect(vm.$el.textContent).toBe('1');
    }).then(done);
  })
```

Reply:
@liximomo Thanks for point out. You are right. I will update the test soon.

Reply:
@liximomo Just updated the test. 

Now we can address the problem of `state`. Using `value` works fine. 

Thanks for the detailed reply.